### PR TITLE
raspiraw: Add command line option to set exposure time in us

### PR DESCRIPTION
**Pull request by fooforever**
Included line readout times in the sesnor modes structure to allow calculation of the exposure register given a requested exposure time

### CODE REVIEW

**Comment by 6by9** [comment]
Patch generally looks fine, but I have a pathological loathing of floats used for no good reason!
Store the line time in nsecs and amend the calc to ```cfg.exposure = (cfg.exposure_us*1000) / sensor_mode->line_time;```. Ideally call it ```line_time_ns``` as well to make it plain the units in use.

I know it only runs once so the optimisation is minimal, but it'll make me happy :-)

**Comment by fooforever** [comment]
Fair enough, though I was keeping to style but didn't think to look for floats being used.

File: host_applications/linux/apps/raspicam/raspiraw.c [hunk 1]
```diff
@@ -676,8 +676,8 @@ int main(int argc, const char** argv) {
 
 	if(cfg.exposure_us != -1)
 	{
-		cfg.exposure = (int)(cfg.exposure_us / sensor_mode->line_time);
-		vcos_log_error("Setting exposure to %d from time %d", cfg.exposure, cfg.exposure_us);
+		cfg.exposure = (int)(cfg.exposure_us * 1000 / sensor_mode->line_time_ns);
```
Comment (6by9) [inline host_applications/linux/apps/raspicam/raspiraw.c]: You no longer need the cast - all variables involved are already ints.
I did give you a duff pointer though - the *1000 has the potential to overflow an int, so you need to cast cfg.exposure_us to int64_t (or long long int) before the multiply.

I've fixed it up, squashed the commits, and merged it all manually. Thanks.

**Comment by 6by9** [comment]
Merged manually - closing.

**Pull request closed without merging**